lib/ConfigFileProcessor.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright (c) 2018, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+
+const { AppOptionsName, DomainTypes, EntityType, EnvOptionsName, LogLevel, OperationType } = require('./Constants');
+const { Endpoints, getConfigTypeError, getCustomNotFoundError, getObjectByOmitting, isEmpty, isNullOrUndefined, readJSON } = require('./Utils');
+
+class ConfigFileProcessor {
+  constructor(options) {
+    this.cliManager = options.cliManager;
+    this.organizationsService = options.organizationsService;
+    this.applicationsService = options.applicationsService;
+    this.environmentsService = options.environmentsService;
+    this.collectionsService = options.collectionsService;
+  }
+
+  _create(source, options, done) {
+    if (source.configType === 'environment') {
```

Comment:
The idea of configType is not to give you a way to fork your code. The idea is to be able to make certain that the user supplied the proper config file for the command. For example, you should not be able to give an environment config file to "kinvey app create".
I can see that the code will throw an error anyway, but we need to give the precise error, something like:
"You have specified the wrong type of config file. Expected: application"